pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1589,28 +1589,37 @@ func (s *AWSCloud) createTags(request *ec2.CreateTagsInput) (*ec2.CreateTagsOutp
 }
 
 func (s *AWSCloud) listSubnetIDsinVPC(vpcId string) ([]string, error) {
-
 	subnetIds := []string{}
 
-	request := &ec2.DescribeSubnetsInput{}
+	sRequest := &ec2.DescribeSubnetsInput{}
 	filters := []*ec2.Filter{}
 	filters = append(filters, newEc2Filter("vpc-id", vpcId))
-	// Note, this will only return subnets tagged with the cluster identifier for this Kubernetes cluster.
-	// In the case where an AZ has public & private subnets per AWS best practices, the deployment should ensure
-	// only the public subnet (where the ELB will go) is so tagged.
 	filters = s.addFilters(filters)
-	request.Filters = filters
+	sRequest.Filters = filters
 
-	subnets, err := s.ec2.DescribeSubnets(request)
+	subnets, err := s.ec2.DescribeSubnets(sRequest)
 	if err != nil {
 		glog.Error("Error describing subnets: ", err)
 		return nil, err
 	}
 
+	rRequest := &ec2.DescribeRouteTablesInput{}
+	rRequest.Filters = filters
```

Comment:
Can we filter using `association.subnet-id`?  I'd like to get this into 1.2 though, so I'm also fine with just a TODO comment here.